File: 프로그래머스/lv2/12911. 다음 큰 숫자/다음 큰 숫자.java
class Solution {
    public int solution(int n) {
        int answer = n+1;


        while (true) {
            String s = Integer.toBinaryString(n);
            int nSize = 0;

            for (int i = 0; i < s.length(); i++) {
                if (s.charAt(i) == '1') nSize++;
            }

            String s2 = Integer.toBinaryString(answer);
            int aSize = 0;

            for (int i = 0; i < s2.length(); i++) {
                if (s2.charAt(i) == '1') aSize++;
            }

            if (nSize == aSize) {
                return answer;
            } else {
                answer++;
            }
        }

    }
}

File: 프로그래머스/2/12911. 다음 큰 숫자/다음 큰 숫자.py
def solution(n):
    answer = 0
    
    a = str(bin(n)[2:]).count("1")
    b = n+1
    c = -1
    while True:
        c = str(bin(b)[2:]).count("1")
        if a == c: return b
        
        b += 1